docs/SPEC.md

```diff
@@ -1228,15 +1250,53 @@ These common values are referred to as the group key value, and can be represent
 
 A tables schema consists of its group key, and its column's labels and types.
 
-[IMPL#463](https://github.com/influxdata/flux/issues/463) Specify the primitive types that make up stream and table types
+A table is represented using the following type:
+
+    type table = {
+        columns: []columnMeta,
+        rows: [...]record,
+    }
+
+Each record must have a property for each column defined on the table.
+
+A schema type is also defined that is useful for inspecing a table's schema ignoring the actual data.
+
+    type schema = { 
+        // List of columns on the table
+        columns: []columnMeta, 
+    }
+
+
+### Channel of tables
 
-### Stream of tables
+A channel represents a potentially unbounded set of tables.
+A channel is grouped into individual tables using the group key.
+Within a channel each table's group key value is unique.
 
-A stream represents a potentially unbounded set of tables.
-A stream is grouped into individual tables using the group key.
-Within a stream each table's group key value is unique.
+    type channel = {
+        tables: [...]table,
+    }
+
+### Stream
+
+A stream represents two channels, a data channel and a meta channel.
+The data channel contains the data being processed.
+The meta channel contains only meta data about the processing of the real data.
+Transformations functions each consume a stream and produce a stream.
+
+The meta channel requires that all tables contain at least a `name` column with a string type.
 
-[IMPL#463](https://github.com/influxdata/flux/issues/463) Specify the primitive types that make up stream and table types
+A stream's type is represented as a set of named channels:
+
+    type stream = {
+        // Data is the raw data to be processed.
+        // Transformations operate on the data channel.
+        data: channel,
+        // Meta is a channel of collected metadata.
+        // Transformations do not operate on the meta channel, 
+        // but they may add tables to it.
+        meta: channel,
+    }
```

Comment:
We should further clarify that there is no total order relationship between data and meta channels.